ui/src/components/admin/ModelsManager/ModelColumns.js

```diff
@@ -42,19 +43,50 @@ const selectedColumns = [
   'updatedBy',
 ];
 
-export const modelStatusPill = (data = null) => {
-  switch (data.status) {
+export const ModelStatusPill = ({ data = null }) => {
+  switch (data) {
     case modelStatus.published:
       return <SmallPill>{modelStatus.published}</SmallPill>;
     case modelStatus.unpublishedChanges:
       return <SmallPill warning>{modelStatus.unpublishedChanges}</SmallPill>;
     case modelStatus.unpublished:
       return <SmallPill info>{modelStatus.unpublished}</SmallPill>;
     default:
-      return <SmallPill primary>{data.status}</SmallPill>;
+      return <SmallPill primary>{data}</SmallPill>;
   }
 };
 
+export const ModelStatusPillWithPublish = ({ name, data }) => {
+  const { location, publishProgress } = useContext(NotificationsContext);
+  const { isPublishingModel, publishRunning } = usePublishNotifications();
+  const [status, setStatus] = useState(null);
+  const publishState = useRef();
+
+  // Force status pill to change to "Published" after single model in bulk publishes
+  // (prior to table data refresh that occurs after entire bulk publish finishes)
+  useEffect(() => {
```

Comment:
If a bulk publish completes while you are on the Model Management table, the data is refreshed to show the latest status of each model (Published, Changes Not Yet Published, Unpublished, etc)

However, with the introduction of the "Publish in Progress" status pull while publishing a model, we can get in the situation where the status pill goes from "Publish in Progress" back to "Changes Not Yet Published" if it finishes publishing before the rest of the bulk completes (and before the user navigates away/refreshes).

This useEffect hook forces the pill to go from "Publish in Progress" to "Published" if the publish completes successfully while the bulk is still ongoing. Note that this is only necessary on the Model Management table, not on the Edit Model page, hence the use of `location.pathname` here.